specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/dataConnectorDefinitions.json

```diff
@@ -0,0 +1,706 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ConnectorDefinitionsAvailability": {
+      "description": "The exposure status of the connector to the customers.",
+      "type": "object",
+      "properties": {
+        "status": {
+          "$ref": "#/definitions/AvailabilityStatus"
+        },
+        "isPreview": {
+          "description": "Gets or sets a value indicating whether the connector is preview.",
+          "type": "boolean"
+        }
+      }
+    },
+    "AvailabilityStatus": {
+      "description": "The exposure status of the connector to the customers. Available values are 0-4 (0=None, 1=Available, 2=FeatureFlag, 3=Internal).",
+      "type": "integer",
+      "format": "int32"
+    },
+    "ConnectivityCriterion": {
+      "description": "The criteria by which we determine whether the connector is connected or not.\r\nFor Example, use a KQL query to check if  the expected data type is flowing).",
+      "required": [
+        "type"
+      ],
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Gets or sets the type of connectivity.",
+          "type": "string"
+        },
+        "value": {
+          "description": "Gets or sets the queries for checking connectivity.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "ConnectorDataType": {
+      "description": "The data type which is created by the connector,\r\nincluding a query indicated when was the last time that data type was received in the workspace.",
+      "required": [
+        "lastDataReceivedQuery",
+        "name"
+      ],
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Gets or sets the name of the data type to show in the graph.",
+          "type": "string"
+        },
+        "lastDataReceivedQuery": {
+          "description": "Gets or sets the query to indicate when relevant data was last received in the workspace.",
+          "type": "string"
+        }
+      }
+    },
+    "DataConnectorDefinition": {
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
+        }
+      ],
+      "required": [
+        "kind"
+      ],
+      "description": "An Azure resource, which encapsulate the entire info requires to display a data connector page in Azure portal,\r\nand the info required to define data connections.",
+      "type": "object",
+      "properties": {
+        "kind": {
```

Comment:
>kind [](http://example.com/codeflow?start=9&length=4)

Tracked resources use "kind" as a top-level property name. Don't use it within the property bag. Maybe connectorKind is suitable.